ethers-core/src/types/transaction/optimism.rs

```diff
@@ -48,14 +48,14 @@ impl DepositTransaction {
         let mut rlp = RlpStream::new();
         rlp.begin_list(NUM_TX_FIELDS);
 
-        rlp_opt(&mut rlp, &self.source_hash);
-        rlp.append(&self.tx.from);
+        rlp.append(&self.source_hash.unwrap_or_default());
+        rlp.append(&self.tx.from.unwrap_or_default());
         rlp_opt(&mut rlp, &self.tx.to);
         rlp_opt(&mut rlp, &self.mint);
-        rlp.append(&self.tx.value);
-        rlp.append(&self.tx.gas);
-        rlp_opt(&mut rlp, &self.is_system_tx);
-        rlp_opt(&mut rlp, &self.tx.data.as_deref());
+        rlp.append(&self.tx.value.unwrap_or_default());
+        rlp.append(&self.tx.gas.unwrap_or_default().as_u64());
+        rlp.append(&self.is_system_tx.unwrap_or_default());
+        rlp.append(&self.tx.data.as_deref().unwrap_or_default());
```

Comment:
I believe rlp_opt was correct here because it falls back to `rlp.append("")` which is equivalent to `rlp.append(&[])`

but this is better